examples/webgl_custom_clipping.html

```diff
@@ -0,0 +1,399 @@
+					mesh.matrixAutoUpdate = false;
+
+					volumeVisualization.add( mesh );
+
+				}
+
+				scene.add( volumeVisualization );
+
+
+				var ground = new THREE.Mesh( planeGeometry,
+						new THREE.MeshPhongMaterial( {
+							color: 0xa0adaf, shininess: 150 } ) );
+				ground.rotation.x = - Math.PI / 2;
+				ground.scale.multiplyScalar( 3 );
+				ground.receiveShadow = true;
+				scene.add( ground );
+
+				// Renderer
+
+				var container = document.body;
+
+				renderer = new THREE.WebGLRenderer();
+				renderer.shadowMap.enabled = true;
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				window.addEventListener( 'resize', onWindowResize, false );
+				container.appendChild( renderer.domElement );
+				// Clipping setup:
+				globalClippingPlanes = createPlanes( GlobalClippingPlanes.length );
+				renderer.clippingPlanes = Empty;
+				renderer.localClippingEnabled = true;
+
+				// Stats
+
+				stats = new Stats();
+				container.appendChild( stats.domElement );
+
+				// Controls
+
+				var controls = new THREE.OrbitControls( camera, renderer.domElement );
+				controls.target.set( 0, 1, 0 );
+				controls.update();
+
+				// GUI
+
+				var gui = new dat.GUI(),
+					folder = gui.addFolder( "Local Clipping" ),
+					props = {
+						get 'Enabled'() { return renderer.localClippingEnabled; },
+						set 'Enabled'( v ) {
+								renderer.localClippingEnabled = v;
+								if ( ! v ) volumeVisualization.visible = false; },
+
+						get 'Shadows'() { return clipMaterial.clipShadows; },
+						set 'Shadows'( v ) { clipMaterial.clipShadows = v; },
+
+						get 'Visualize'() { return volumeVisualization.visible; },
+						set 'Visualize'( v ) {
+								if ( renderer.localClippingEnabled )
+									volumeVisualization.visible = v; }
+					};
+
+				folder.add( props, 'Enabled' );
+				folder.add( props, 'Shadows' );
+				folder.add( props, 'Visualize' ).listen();
+
+				gui.addFolder( "Global Clipping" ).
+						add( {
+							get 'Enabled'() { return renderer.clippingPlanes !== Empty; },
+							set 'Enabled'( v  ) { renderer.clippingPlanes = v ?
+									globalClippingPlanes : Empty; }
+						}, "Enabled" );
+
+
+				// Start
+
+				startTime = Date.now();
+
+			}
+
+			function onWindowResize() {
+
+				camera.aspect = window.innerWidth / window.innerHeight;
+				camera.updateProjectionMatrix();
+
+				renderer.setSize( window.innerWidth, window.innerHeight );
+
+			}
+
+			var transform = new THREE.Matrix4(),
+				tmpMatrix = new THREE.Matrix4();
+
+			function animate() {
+
+				var currentTime = Date.now(),
+					time = ( currentTime - startTime ) / 1000;
+
+				requestAnimationFrame( animate );
+
+				transform.makeTranslation( 0, 1, 0 ).
+						multiply( tmpMatrix.makeRotationX( time * 0.5 ) ).
+						multiply( tmpMatrix.makeRotationY( time * 0.2 ) );
+
+				object.matrix.copy( transform );
+				object.updateMatrixWorld( true );
```

Comment:
> We had many debates about what the user should and should-not be able to do years ago, so there is a lot of history regarding this issue. New users often don't know what they are doing. The policy has been to advise against setting the object matrix directly. That is the renderer's job.

OK, I get that. I placed this example under "advanced". It's probably a bad choice for a 3D newbie to study. Then OTOH, the tricky bits are packaged for "faithful copy & paste". 

In some cases you need the matrices, and I believe it might be difficult to get away without in this one. There is a matrix interface, and of course I want to use it in a "generally advisable" fashion.
 
eadd4df gets rid of `.matrixAutoUpdate = false` and the transform of the main mesh is now calculated setting `.position` and `.rotation`. I want to apply the same translation / rotation to the clipping volume (an array of `Plane` objects) - and therefore I need a matrix, so I make sure the matrix is up to date and copy it ([see code](https://github.com/tschw/three.js/blob/ClippingPlanes/examples/webgl_custom_clipping.html#L362-L374)).

> That being said, I have no problem revisiting this issue if you feel you have a compelling reason to do so.

For visualizing the (world space) `Plane` objects as they are, I built a [utility function](https://github.com/tschw/three.js/blob/ClippingPlanes/examples/webgl_custom_clipping.html#L92-L129) that turns a plane into the (world) matrix for a `Mesh` with `PlaneGeometry`. It's fairly easy to do: An orthonormal space is constructed around `.normal` for the Z-Axis. [Another utility function](https://github.com/tschw/three.js/blob/ClippingPlanes/examples/webgl_custom_clipping.html#L131-L139) is provided to then align an object according to the matrix.

I can't really think of simpler means to do this stuff. In particular, I believe it would be ill-advised to encourage the user to use Euler angles for world space stuff, because they become unmanageable once there's a bit of scene hierarchy.